src/main/groovy/life/qbic/utils/DataParserException.groovy

```diff
@@ -0,0 +1,22 @@
+package life.qbic.utils
+
+/**
+ * Exception that shall be thrown if parsing of a data structures fail.
+ *
+ * @since 1.7.0
+ */
+class DataParserException extends RuntimeException {
```

Comment:
```suggestion
class DataParserException <T> extends RuntimeException {
```
It might be helpful to have the exception also be typed. What do you think?

Reply:
Hm, what problem does it solve? The exception does not return type specific information, but is merely a general data parser exception.
Can you elaborate the use case of a typed exception?

Reply:
If we introduce a generic field here we could pass an extra argument of what exactly failed to parse. For example, a parser tries to parse the wrong type of dataset. In the calling method, we could catch the exception and retain the information that the parser assumed. That could be helpful in case we have multiple parsers and we catch the exceptions in a collected fashion. The described problem can also be solved differently, we have to make sure to avoid the def for that then so that we know what type of parser threw the exception.